src/v/redpanda/application.cc

```diff
@@ -355,41 +367,55 @@ void application::shutdown() {
      * possible.
      */
     if (_datalake_coordinator_mgr.local_is_initialized()) {
-        _datalake_coordinator_mgr
-          .invoke_on_all(&datalake::coordinator::coordinator_manager::shutdown)
-          .get();
+        shutdown_with_watchdog(_datalake_coordinator_mgr, [](auto& mgr) {
+            return mgr.invoke_on_all(
+              &datalake::coordinator::coordinator_manager::shutdown);
+        });
     }
     if (_datalake_manager.local_is_initialized()) {
-        _datalake_manager.invoke_on_all(&datalake::datalake_manager::shutdown)
-          .get();
+        shutdown_with_watchdog(_datalake_manager, [](auto& mgr) {
+            return mgr.invoke_on_all(&datalake::datalake_manager::shutdown);
+        });
     }
     if (_datalake_credential_mgr.local_is_initialized()) {
-        _datalake_credential_mgr
-          .invoke_on_all(&datalake::credential_manager::stop)
-          .get();
+        shutdown_with_watchdog(_datalake_credential_mgr, [](auto& mgr) {
+            return mgr.invoke_on_all(&datalake::credential_manager::stop);
+        });
     }
 
     // Stop all partitions before destructing the subsystems (transaction
     // coordinator, etc). This interrupts ongoing replication requests,
     // allowing higher level state machines to shutdown cleanly.
     if (partition_manager.local_is_initialized()) {
-        partition_manager
-          .invoke_on_all(&cluster::partition_manager::stop_partitions)
-          .get();
+        shutdown_with_watchdog(partition_manager, [](auto& partition_manager) {
+            return partition_manager.invoke_on_all(
+              &cluster::partition_manager::stop_partitions);
+        });
     }
 
     // Wait for all requests to finish before destructing services that may be
     // used by pending requests.
     if (_kafka_server.ref().local_is_initialized()) {
-        _kafka_server.wait_for_shutdown().get();
-        _kafka_server.stop().get();
+        shutdown_with_watchdog(_kafka_server, [](auto& kafka_server) {
+            return kafka_server.wait_for_shutdown();
+        });
+        shutdown_with_watchdog(_kafka_server, [](auto& kafka_server) {
+            return kafka_server.stop();
+        });
```

Comment:
how will these two be distinguished in the logs?

Reply:
a log line number

Reply:
@mmaslankaprv i don't think that's true. i think they'll all have the same line number as the log line in the template:

```
nwatkins@acid:~/src/redpanda$ cat data/node0/redpanda.log | grep "Shutting down" | grep server_app
INFO  2025-07-28 08:59:16,252 [shard 0:main] main - application.h:359 - Shutting down: kafka::server_app
INFO  2025-07-28 08:59:16,273 [shard 0:main] main - application.h:359 - Shutting down: kafka::server_app
INFO  2025-07-28 08:59:16,274 [shard 0:main] main - application.h:359 - Shutting down: kafka::server_app
```